### PR TITLE
Fix amqp message consumption

### PR DESCRIPTION
**Pull request by arbulu89**
Fix amqp messages consumption for fact gathered event

### CODE REVIEW

File: lib/wanda/messaging/adapters/amqp/consumer.ex [hunk 1]
```diff
@@ -25,10 +25,10 @@ defmodule Wanda.Messaging.Adapters.AMQP.Consumer do
 
   @impl GenRMQ.Consumer
   def handle_message(%GenRMQ.Message{payload: payload} = message) do
-    case Contracts.from_event(payload) do
-      {:ok, event} ->
-        Policy.handle_event(event)
-
+    with {:ok, event} <- Contracts.from_event(payload),
```
Comment (arbulu89) [inline lib/wanda/messaging/adapters/amqp/consumer.ex:28]: Some doubts with this `with` XD

**Review by fabriziosestito — APPROVED** [review]

File: lib/wanda/execution.ex [hunk 2]
```diff
@@ -17,15 +17,21 @@ defmodule Wanda.Execution do
...
+            group_id: group_id,
+            targets: targets,
+            checks: checks,
+            config: config}
+         ) do
+      {:ok, _} ->
+        :ok
+
+      {:error, _} = error ->
+        error
```
Comment (dottorblaster) [inline lib/wanda/execution.ex:33]: Why weren't we happy with the previous one?

**Review by dottorblaster — APPROVED** [review]
LGTM

File: lib/wanda/execution.ex (hunk 2 above)
Comment (fabriziosestito) [reply lib/wanda/execution.ex:33]: execution behaviour is defined to return :ok or error, returning a pid is an implementation detail of this specific implementation. we are just interested to know that the execution was started properly

File: lib/wanda/messaging/adapters/amqp/consumer.ex (hunk 1 above)
Comment (fabriziosestito) [reply lib/wanda/messaging/adapters/amqp/consumer.ex:28]: that's ok, someone does not like using else with `with` blocks but I find it too strict

Comment (dottorblaster) [reply lib/wanda/messaging/adapters/amqp/consumer.ex:28]: `someone` is me but I think it's ok, no big deal

Comment (fabriziosestito) [reply lib/wanda/messaging/adapters/amqp/consumer.ex:28]: it depends. if you have several `else` most probably you are better using other constructs, here we have a default error case that it does not just return but needs to call an error handling function

File: lib/wanda/execution.ex (hunk 2 above)
Comment (arbulu89) [reply lib/wanda/execution.ex:33]: Glad to see that I had the same solution mind XD

**Review by arbulu89 — COMMENTED** [review]
Thanks @fabriziosestito

**Pull request merged**